inc/classes/admin/settings/class-settings.php

```diff
@@ -496,4 +490,42 @@ function( $error ) {
 	public function sanitize_checkbox( $value ) {
 		return isset( $value ) ? 1 : 0;
 	}
+
+	/**
+	 * Sanitizes the DNS Prefetch sub-option value
+	 *
+	 * @since 3.5.1
+	 * @author Remy Perona
+	 *
+	 * @param array $input Array of values for the WP Rocket settings option.
+	 * @return array Sanitized array for the DNS Prefetch sub-option
+	 */
+	private function sanitize_dns_prefetch( $input ) {
+		if ( ! isset( $input['dns_prefetch'] ) || empty( $input['dns_prefetch'] ) ) {
+			return [];
+		}
+
+		$value = $input['dns_prefetch'];
+
+		if ( ! is_array( $value ) ) {
+			$value = explode( "\n", $value );
+		}
+
+		$value = array_filter( array_map( 'trim', $value ) );
+
+		if ( empty( $value ) ) {
+			return [];
+		}
+
+		$value = array_filter( array_map( 'esc_url_raw', $value ) );
```

Comment:
@Tabrisrp This works fine. But it's less performant as it takes more memory and cycles. Why?

Each time `array_map` runs, it has to iterate through all of the values. Plus it creates a new array in memory. So it's Big O for time-complexity is N and space-complexity is also N.

Same with `array_filter`. 

Think about that. There are 2 `array_maps` and `array_filters`. That's 4 times iterating through all of the values and creating 4 new arrays in memory.

We can improve this by using a `foreach`:

```php
$urls = [];
foreach ( $values as $index => $url ) {
	$url = trim( $url );
	if ( empty( $url ) ) {
		continue;
	}
	$url = esc_url_raw( $url );
	if ( empty( $url ) ) {
		continue;
	}
	$urls[] = $url;
}

if ( empty( $urls ) ) {
	return [];
}
```

Notice:

1. It only iterates through the values 1x instead of 4x.
2. It creates 1 array instead of 4. Plus, the array is sized for only the values we need.
3. It's still readable and understandable.